### PR TITLE
clang-tidy-12

### PR DESCRIPTION
**Pull request by pattivacek**
Currently includes https://github.com/uptane/aktualizr/pull/38.

### CODE REVIEW

**Comment by pattivacek** [comment]
FYI @cajun-rat @mike-sul if https://github.com/uptane/aktualizr/pull/38 wasn't enough here's this, too. Pretty small, actually.

File: src/libaktualizr/storage/sqlstorage_base.h [hunk 1]
```diff
@@ -13,12 +13,12 @@ enum class DbVersion : int32_t { kEmpty = -1, kInvalid = -2 };
 class StorageLock {
  public:
   StorageLock() = default;
-  StorageLock(boost::filesystem::path path);
+  explicit StorageLock(boost::filesystem::path path);
+  virtual ~StorageLock();
   StorageLock(StorageLock &other) = delete;
-  StorageLock &operator=(StorageLock &other) = delete;
   StorageLock(StorageLock &&other) = default;
+  StorageLock &operator=(StorageLock &other) = delete;
```
Comment (mike-sul) [inline src/libaktualizr/storage/sqlstorage_base.h:20]: **const** StorageLock &other?

File: src/aktualizr_secondary/secondary_tcp_server.h [hunk 2]
```diff
@@ -23,11 +23,12 @@ class SecondaryTcpServer {
 
   SecondaryTcpServer(MsgHandler& msg_handler, const std::string& primary_ip, in_port_t primary_port, in_port_t port = 0,
                      bool reboot_after_install = false);
-
+  ~SecondaryTcpServer() = default;
   SecondaryTcpServer(const SecondaryTcpServer&) = delete;
+  SecondaryTcpServer(SecondaryTcpServer&&) = delete;
   SecondaryTcpServer& operator=(const SecondaryTcpServer&) = delete;
+  SecondaryTcpServer& operator=(const SecondaryTcpServer&&) = delete;
```
Comment (mike-sul) [inline src/aktualizr_secondary/secondary_tcp_server.h:30]: No need `const` for the assignment move operator?

Comment (pattivacek) [reply src/aktualizr_secondary/secondary_tcp_server.h:30]: Correct, it actually shouldn't be `const`. If you're deleting it, gcc/clang/clang-tidy don't seem to care, but the "proper" definition is without.

File: src/libaktualizr/storage/sqlstorage_base.h (hunk 1 above)
Comment (pattivacek) [reply src/libaktualizr/storage/sqlstorage_base.h]: Apparently doesn't matter if it's deleted, but for consistency, I've fixed it.

**Review by mike-sul — APPROVED** [review]

**Pull request merged**